File: front/lib/models/favorite_model.dart
// lib/models/favorite_model.dart

class FavoriteModel {
  final String userId;
  final String placeId;

  FavoriteModel({required this.userId, required this.placeId});

  factory FavoriteModel.fromJson(Map<String, dynamic> json) {
    return FavoriteModel(
      userId: json['userId'],
      placeId: json['placeId'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'userId': userId,
      'placeId': placeId,
    };
  }
}

File: backend/models/favorite_model.js
// backend/models/favorite_model.js

const mongoose = require('mongoose');
const Schema = mongoose.Schema;

const favoriteSchema = new Schema(
  {
    userId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: 'AUTH',
      required: true
    },
    placeId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: 'Place',
      required: true
    },
  },
  {
    timestamps: true
  }
);

const Favorite = mongoose.model('Favorite', favoriteSchema);
module.exports = Favorite;
